tests/AuthService.test.js

```diff
@@ -1,12 +1,78 @@
-const AuthService = require ("../core/services/AuthService.js");
+global.AuthService = require ("../core/services/AuthService.js");
+global._ = require("lodash");
 
 describe ("AuthService", () => {
-  it ("Password encrypt and check", () =>{
+  it("Password encrypt and check", () => {
     const passwords = ["rrr", "fdgfdg", 1234, {a: "b"}];
     for (let password of passwords) {
       (typeof(password) === "string")
       ? expect(AuthService.checkPassword(password, AuthService.encryptPassword(password))).toBeTruthy()
       : expect(() => {AuthService.encryptPassword(password)}).toThrow();
     }
-  })
+  });
+  it("Token generate and check", () => {
+    const operator = {
+      id: 23,
+      email: "example@gmail.com",
+      passwordHash: AuthService.encryptPassword("heDK4src4w")
+    };
+    const token = AuthService.generateToken(operator);
+    const decodedToken = AuthService.checkAndDecodeToken(token);
+    expect(decodedToken.id).toBe(operator.id);
+    expect(decodedToken.email).toBe(operator.email);
+    expect(decodedToken.exp).toBeGreaterThan(Date.now()/1000);
```

Comment:
Well, that test is not strong enough. Our function `generateToken` should pick only required attributes. If this feature is broken (all the attributes go to client), the test still passes. On the other hand, if we later decide to let some additional attributes (like rights) into the client, and later this feature is broken, the test will pass.

So, the right way is: 
* Have somewhere a constant containing an array of picked attributes (e.g. AuthService.OPERATOR_ATTRIBUTES_FOR_JWT);
* Use this constant from the `generateToken` function and from this test;
* So, in test it's better to create an object containing expected result – a subset of operator attributes using `_.pick`, just like in `generateToken`.
* Then, add into this object expected value of `exp`. To get it, you can manually decode the token right here – in test, pick the `iat` value, and add seconds from the config.
* Then, compare expected result with the one returned from `checkAndDecodeToken` using `toStrictEqual`